netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -1105,4 +1209,245 @@ protected final void fireProtocolNegotiationEvent(ChannelHandlerContext ctx) {
       ctx.fireUserEventTriggered(pne);
     }
   }
+
+  static final class SslEngineWrapper extends SSLEngine {
```

Comment:
This is a lot of boilerplate. It'd be good to move these to their own files. You can have it just be the "wrapper", and then still override `getPeerHost()`/`getHandshakeSession()` in the file here.

It's interesting that only `getSSLParameters()` uses `sslEngine`. Maybe the separate file could be a `NoopSslEngine`?

Reply:
Done.